test/jdk/jdk/incubator/vector/TestMaskedMacroLogicVector.java

```diff
@@ -0,0 +1,493 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8273322
+ * @modules jdk.incubator.vector
+ * @requires vm.compiler2.enabled
+ * @library /test/lib
+ * @run main/othervm -XX:+IgnoreUnrecognizedVMOptions
+ *           -Xbatch -XX:-TieredCompilation -XX:CICompilerCount=1 -XX:UseAVX=3
+ *           -XX:CompileCommand=quiet -XX:+TraceNewVectors
+ *           test.jdk.incubator.TestMaskedMacroLogicVector
+ */
+
+package test.jdk.incubator;
+
+import java.util.Random;
+import java.util.concurrent.Callable;
+import jdk.incubator.vector.*;
+
+public class TestMaskedMacroLogicVector {
```

Comment:
This test looks a strong a candidate to use the IR test framework.

https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/lib/ir_framework/README.md 

Can we use that? 

IIUC this PR is an additional optimization rather than fundamental support for masking. In that case I think we have more time to explore better testing, and this could be a follow on PR from the main masking that will soon be under review.